// ngx-quill/esm5/lib/quill.service.js
import { __awaiter, __decorate, __generator, __param } from "tslib";
import { Injectable, Inject } from '@angular/core';
import { QUILL_CONFIG_TOKEN } from './quill-editor.interfaces';
import * as i0 from "@angular/core";
import * as i1 from "./quill-editor.interfaces";
var QuillService = /** @class */ (function () {
    function QuillService(config) {
        this.config = config;
        this.count = 0;
    }
    QuillService.prototype.getQuill = function () {
        var _this = this;
        this.count++;
        if (!this.Quill && this.count === 1) {
            this.$importPromise = new Promise(function (resolve) { return __awaiter(_this, void 0, void 0, function () {
                var _a;
                var _this = this;
                var _b, _c;
                return __generator(this, function (_d) {
                    switch (_d.label) {
                        case 0:
                            _a = this;
                            return [4 /*yield*/, import('quill')];
                        case 1:
                            _a.Quill = ((_d.sent()).default);
                            // Only register custom options and modules once
                            (_b = this.config.customOptions) === null || _b === void 0 ? void 0 : _b.forEach(function (customOption) {
                                var newCustomOption = _this.Quill.import(customOption.import);
                                newCustomOption.whitelist = customOption.whitelist;
                                _this.Quill.register(newCustomOption, true, _this.config.suppressGlobalRegisterWarning);
                            });
                            (_c = this.config.customModules) === null || _c === void 0 ? void 0 : _c.forEach(function (_a) {
                                var implementation = _a.implementation, path = _a.path;
                                _this.Quill.register(path, implementation, _this.config.suppressGlobalRegisterWarning);
                            });
                            resolve(this.Quill);
                            return [2 /*return*/];
                    }
                });
            }); });
        }
        return this.$importPromise;
    };
    QuillService.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: [QUILL_CONFIG_TOKEN,] }] }
    ]; };
    QuillService.ɵprov = i0.ɵɵdefineInjectable({ factory: function QuillService_Factory() { return new QuillService(i0.ɵɵinject(i1.QUILL_CONFIG_TOKEN)); }, token: QuillService, providedIn: "root" });
    QuillService = __decorate([
        Injectable({
            providedIn: 'root'
        }),
        __param(0, Inject(QUILL_CONFIG_TOKEN))
    ], QuillService);
    return QuillService;
}());
export { QuillService };
